ror-tools/src/test/scala/tech/beshu/ror/tools/RorToolsAppSuite.scala

```diff
@@ -0,0 +1,231 @@
+        """Error: Path [/wrong_directory] does not exist
+          |Try --help for more information.""".stripMargin
+      )
+    }
+    "Successfully patch, verify and unpatch" in withFreshEsDirectory { () =>
+      // Patch
+      val hashBeforePatching = DirectoryUtils.calculateHash(esLocalPath)
+      val (patchResult, patchOutput) = captureResultAndOutput {
+        RorToolsAppHandler.handle(Array("patch", "--I_UNDERSTAND_AND_ACCEPT_ES_PATCHING", "yes", "--es-path", esLocalPath.toString))(_)
+      }
+      patchResult should equal(Result.Success)
+      patchOutput should include(
+        """Checking if Elasticsearch is patched ...
+          |Creating backup ...
+          |Patching ...
+          |Elasticsearch is patched! ReadonlyREST is ready to use"""
+          .stripMargin
+      )
+      val hashAfterPatching = DirectoryUtils.calculateHash(esLocalPath)
+
+      // Verify
+      val (verifyResult, verifyOutput) = captureResultAndOutput {
+        RorToolsAppHandler.handle(Array("verify", "--es-path", esLocalPath.toString))(_)
+      }
+
+      verifyResult should equal(Result.Success)
+      verifyOutput should include(
+        """Checking if Elasticsearch is patched ...
+          |Elasticsearch is patched! ReadonlyREST can be used"""
+          .stripMargin
+      )
+
+      // Unpatch
+      val hashBeforeUnpatching = DirectoryUtils.calculateHash(esLocalPath)
+      val (unpatchResult, unpatchOutput) = captureResultAndOutput {
+        RorToolsAppHandler.handle(Array("unpatch", "--es-path", esLocalPath.toString))(_)
+      }
+      unpatchResult should equal(Result.Success)
+      unpatchOutput should include(
+        """Checking if Elasticsearch is patched ...
+          |Restoring ...
+          |Elasticsearch is unpatched! ReadonlyREST can be removed now"""
+          .stripMargin
+      )
+      val hashAfterUnpatching = DirectoryUtils.calculateHash(esLocalPath)
+
+      hashBeforePatching should equal(hashAfterUnpatching)
+      hashAfterPatching should equal(hashBeforeUnpatching)
+      hashBeforePatching shouldNot equal(hashAfterPatching)
+    }
+  }
+
+  private def captureResultAndOutput(block: InOut => Result): (Result, String) = {
+    val inOut = new CapturingOutputAndMockingInput()
+    val result = block(inOut)
+    (result, inOut.getOutputBuffer)
+  }
+
+  private def captureResultAndOutputWithInteraction(block: InOut => Result, response: String): (Result, String) = {
+    val inOut = new CapturingOutputAndMockingInput(Some(response))
+    val result = block(inOut)
+    (result, inOut.getOutputBuffer)
+  }
+
+  private def withTestEsContainer(withStartedEs: EsContainer => Unit): Unit = {
+    val manager = new TestEsContainerManager
+    try {
+      manager.start()
+        .map(_ => withStartedEs(manager.esContainer))
+        .runSyncUnsafe(5 minutes)
+    } finally {
+      manager.stop().runSyncUnsafe()
+    }
+  }
+
+  private def unTar(tarPath: Path, outputPath: Path): Unit = {
```

Comment:
let's move the method to some utils? Please review existing utils objects/classes

Reply:
Moved to FileUtils in this module.